neuro_san/coded_tools/web_search.py

```diff
@@ -28,6 +28,14 @@ class WebSearch(CodedTool):
     def __init__(self):
         self.top_n = 5
 
+    async def async_invoke(self, args: Dict[str, Any], sly_data: Dict[str, Any]) -> Union[Dict[str, Any], str]:
+        """
+        As much as we'd prefer an asynchronous entry point, the code below uses synchronous
+        calls for a simple example. Asynchronous calls make for a more performant server
+        environment.
+        """
+        raise NotImplementedError
```

Comment:
Move async_invoke() non-implementation to the top with a comment about what's nicer.